docs/howto/secrets.md

```diff
@@ -1,28 +1,76 @@
+This information is stored by the operator as secrets after resource creation.
 
 The operator provides two options to store these secrets:
 
-1. **Kubernetes secrets**: This is the default option. If the secretname is not specified in the Spec, a secret with the same name as  ObjectMeta.name is created.
+1. **Kubernetes secrets**: A Kubernetes secret will be created alongside the resource. This is the default option. For details about how the secrets are named, see [Secret naming](#secret-naming).
 
-2. **Azure Keyvault secrets**: You can specify the name of the Azure Keyvault to use to store secrets through the environment variable `AZURE_OPERATOR_KEYVAULT`.
+2. **Azure Key Vault secrets**: You can specify the default Azure Key Vault to store secrets in through via the `AZURE_OPERATOR_KEYVAULT` field in the `azureoperatorsettings` secret.
+This can be set by exporting it as an environment variable and then including it in the secret creation.
+    ```
+    export AZURE_OPERATOR_KEYVAULT=mykeyvault
+    kubectl create secret generic azureoperatorsettings \
+            ... (other settings) \
+            --from-literal=AZURE_OPERATOR_KEYVAULT=${AZURE_OPERATOR_KEYVAULT}
+    ```
+    Or if you're using Helm:
+    ```
+    helm upgrade --install aso aso/azure-service-operator \
+            --create-namespace \
+            --namespace=azureoperator-system \
+            ... \
+            --set azureOperatorKeyvault=$AZURE_OPERATOR_KEYVAULT
+    ```
 
-If the secretname is not specified in the Spec, the secret in Keyvault is normally created with the name `<kubernetes-namespace-of=resource>-<ObjectMeta.name>`. The namespace is preprended to the name to avoid name collisions across Kubernetes namespaces.
+For details about how the secrets are named, see [Secret naming](#secret-naming).
 
+Some things to note about the Key Vault you use with the operator:
+1. The KeyVault should have an access policy added for the identity under which the Operator runs as.
+   This access policy should include  at least `get`, `set`, `list` and `delete` Secret permissions.
+2. You can use a Key Vault with "Soft delete" enabled. However, you cannot use a Key Vault with "Purge Protection" enabled, as this prevents the
+   secrets from being deleted and causes issues if a resource with the same name is re-deployed.
 
-```
-export AZURE_OPERATOR_KEYVAULT=OperatorSecretKeyVault
-```
+## Secret naming
+
+There are two versions of secret naming used by the Azure Service Operator. The secret naming version is controlled by the `AZURE_SECRET_NAMING_VERSION` field of the `azureoperatorsettings` secret.
+Valid values are `1` and `2`. Version `2` is the default when installing the operator via Helm. If no value is specified for `AZURE_SECRET_NAMING_VERSION` the default is version `1` for backwards compatibility.
```

Comment:
As mentioned above, wondering if I should change this so that the default is 2 in code as well.